virtio-devices/src/balloon.rs

```diff
@@ -94,6 +95,42 @@ pub struct VirtioBalloonConfig {
     actual: u32,
 }
 
+#[derive(Copy, Clone, Debug)]
+struct PartPage {
+    addr: u64,
+    bitmap: u64,
+}
+
+impl PartPage {
+    fn new() -> Self {
+        Self { addr: 0, bitmap: 0 }
+    }
+
+    fn pfn_match(&self, addr: u64) -> bool {
+        // SAFETY: FFI call. Trivially safe.
+        let page_size: u64 = unsafe { sysconf(_SC_PAGESIZE) as u64 };
+        self.addr == addr & !(page_size - 1)
+    }
+
+    fn bitmap_full(&mut self, page_size: u64) -> bool {
+        let bp_num_per_page = page_size >> VIRTIO_BALLOON_PFN_SHIFT;
+        self.bitmap & ((1 << bp_num_per_page) - 1) == (1 << bp_num_per_page) - 1
```

Comment:
When the page size is larger than 256K, you will have a overflow from the shit left operation.

Reply:
OK